### PR TITLE
fix(dedicated:server): ipmi IP check in US

### PR DESCRIPTION
**Pull request by lizardK**

### CODE REVIEW

File: client/app/dedicated/server/dedicated-server.service.js [hunk 1]
```diff
@@ -688,17 +688,24 @@ angular
         };
 
         this.ipmiStartConnection = function (serviceName, type, ttl, ipToAllow, sshKey) {
-            return OvhHttp.post("/dedicated/server/{serviceName}/features/ipmi/access", {
+            let promise = $q.when(ipToAllow);
+
+            if (constants.target === "US") {
+                promise = this.getIpGeolocation().then((data) => data.ip);
```
Comment (jleveugle) [inline client/app/dedicated/server/dedicated-server.service.js]: You should do this too :

```javascript
this.getIpGeolocation().then(({ ip ) => ip);
```

File: client/app/dedicated/server/dedicated-server.service.js [hunk 2]
```diff
@@ -687,18 +687,25 @@ angular
             });
         };
 
-        this.ipmiStartConnection = function (serviceName, type, ttl, ipToAllow, sshKey) {
-            return OvhHttp.post("/dedicated/server/{serviceName}/features/ipmi/access", {
+        this.ipmiStartConnection = function ({ serviceName, type, ttl, ipToAllow, sshKey, withGeolocation }) {
+            let promise = $q.when(ipToAllow);
+
+            if (withGeolocation) {
+                promise = this.getIpGeolocation().then((data) => _.get(data, "ip"));
```
Comment (antleblanc) [inline client/app/dedicated/server/dedicated-server.service.js]: As @jleveugle said you can write also do:

```js
promise = this.getIpGeolocation().then(({ ip }) => ip);
```

File: client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js [hunk 3]
```diff
@@ -244,8 +249,14 @@ angular.module("App").controller("ImpiCtrl", ($scope, $translate, Server, Pollin
         $scope.loader.javaReady = false;
         $scope.loader.javaLoading = true;
         $scope.loader.buttonStart = true;
-
-        Server.ipmiStartConnection($stateParams.productId, "kvmipJnlp", $scope.ttl, $scope.ipmi.model.clientIp).then(
+        const withGeolocation = _.contains(["HIL_1", "VIN_1"], $scope.server.datacenter) && constants.target === "US";
```
Comment (Jisay) [inline client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js]: `_.contains` -> `_.includes`

File: client/app/dedicated/server/dedicated-server.service.js [hunk 4]
```diff
@@ -687,18 +687,25 @@ angular
...
+            }
+
+            return promise.then((ip) => OvhHttp.post("/dedicated/server/{serviceName}/features/ipmi/access", {
```
Comment (antleblanc) [inline client/app/dedicated/server/dedicated-server.service.js]: You can also rename your param to: `ipToAllow`

File: client/app/dedicated/server/dedicated-server.service.js [hunk 5]
```diff
@@ -687,18 +687,25 @@ angular
...
                 rootPath: "apiv6",
                 urlParams: {
                     serviceName
                 },
                 data: {
                     ttl,
                     type,
-                    sshKey
+                    sshKey,
+                    ipToAllow: ip
```
Comment (antleblanc) [inline client/app/dedicated/server/dedicated-server.service.js]: If you param has been renamed (see previous comment), you can use shorthand:

```js
{
  …
  sshKey,
  ip
}
```

File: client/app/dedicated/server/dedicated-server.service.js (hunk 4 above)
Comment (lizardK) [reply client/app/dedicated/server/dedicated-server.service.js]: nope : 
error  'ipToAllow' is already declared in the upper scope  no-shadow

File: client/app/dedicated/server/dedicated-server.service.js (hunk 5 above)
Comment (lizardK) [reply client/app/dedicated/server/dedicated-server.service.js]: error  'ipToAllow' is already declared in the upper scope  no-shadow

**Pull request merged**